tests/testthat/test-semicolon_linter.R

```diff
@@ -69,9 +69,46 @@ test_that("Compound semicolons only", {
 
 
 test_that("Trailing semicolons only", {
-  linter <- semicolon_terminator_linter(semicolon  =  "trailing")
+  linter <- semicolon_linter(allow_compound = TRUE)
   expect_lint("a <- 1;b <- 2", NULL, linter)
   expect_lint("function() {a <- 1;b <- 2}\n", NULL, linter)
   expect_lint("f <-\n 1 ;f <- 1.23", NULL, linter)
   expect_lint("function(){\nf <-\n 1 ;f <- 1.23\n}", NULL, linter)
 })
+
+test_that("deprecation notices for semicolon_terminator_linter succeed, and the deprecated version works", {
+  expect_warning(
+    linter <- semicolon_terminator_linter(),
+    "Function semicolon_terminator_linter was deprecated",
+    fixed = TRUE
+  )
+  expect_lint("a <- 1", NULL, linter)
+  expect_lint("a <- 1;", rex::rex("Trailing semicolons are not needed."), linter)
+  expect_lint("a <- 1; b <- 2", rex::rex("Compound semicolons are not needed."), linter)
```

Comment:
The lint message seems a bit off. Not needed -> discouraged?
They aren't redundant after all.